### PR TITLE
concurrent utilities / it-academy.by / M-JC1-66-20 / Anastasia Gawriljuk / Done projects 'Master-workers', 'Improved Parallel Calculator', 'Censor'

### PR DESCRIPTION
**Pull request by Semargl92**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/censor/FileCollector.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                        try {
+                            throw new CensorException("Collector: Some files in this directory contains no text");
+                        } catch (CensorException exception) {
+                            System.out.println(exception.getMessage());
+                        }
+                    }
+                }
+            }
+            System.out.println("Files were found - " + chosenFiles.size());
+        } else {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/censor/FileCollector.java:46]: Best way: use Executor for this job.

**Review by rakovets — CHANGES_REQUESTED** [review]
Good job but you need to fix the following problems.

File: src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/improved_parallel_calculator/ImprovedParallelCalculator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                }
+            System.out.println(marker + " - " + (System.currentTimeMillis() - start));
+        };
+
+        ExecutorService executor = Executors.newFixedThreadPool(numberOfThreads);
+        for (int x = 1; x <= numberOfThreads; x++) {
+            executor.execute(calculatorUnit);
+        }
+        executor.shutdown();
+        return result;
```
[the remark this answers is not in the document]
Comment (Semargl92) [reply src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/improved_parallel_calculator/ImprovedParallelCalculator.java:50]: Interface 'ExecutorService' does not have methods with such signatures, so I used 'invokeAll' instead to fix the problem.

File: src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/censor/FileCollector.java (hunk 1 above)
Comment (Semargl92) [reply src/main/java/com/rakovets/course/java/core/practice/concurrent/utilities/censor/FileCollector.java:46]: Dima, explain, please, what do you mean. Class 'FileCollector' already implements Callable and is submitted in Class 'Censor' by ExecutorService. Or do you suggest to make multithreaded process of checking files in directory, too?

**Review by rakovets — COMMENTED** [review]
Good job

**Pull request closed without merging**